src/main/ui/cheat-code-detector.gd

```diff
@@ -0,0 +1,36 @@
+class_name CheatCodeDetector
+extends Node
+
+signal cheat_detected
+
+const MAX_LENGTH = 32
+const CODE_KEYS := {
+	"a":"", "b":"", "c":"", "d":"", "e":"", "f":"", "g":"", "h":"", "i":"",
+	"j":"", "k":"", "l":"", "m":"", "n":"", "o":"", "p":"", "q":"", "r":"",
+	"s":"", "t":"", "u":"", "v":"", "w":"", "x":"", "y":"", "z":""
```

Comment:
Could we expand this to include number keys, the space bar, or other things like that?

Reply:
Now includes all numbers and lower-case symbols like the hyphen key. Doesn't include upper-case or shift-symbols like the underscore key.